### PR TITLE
Gitignore exception

### LINKED ISSUE #423: Validation checks on NewRepository.GitignoreTemplate
One of our integration tests was failing with this response:

```
{
  "message":"Validation Failed",
  "documentation_url":"http://developer.github.com/v3/repos/#create",
  "errors":
  [{
    "resource":"Repository",
    "code":"custom",
    "field":"gitignore_template",
    "message":"gitignore_template is an unknown gitignore template."
  }]
}
```

This is because we were passing `visualstudio`, rather than `VisualStudio`.

We should surface this as a proper exception message in the future...

### PR DESCRIPTION
**Pull request by willsb**
Fixes #423

### CODE REVIEW

**Comment by Haacked** [comment]
Ok, the fact that it always says `gitignore_template` at the beginning of the string appears to be a bug in our API. Let's not rely on the bug behavior.

**Comment by willsb** [comment]
Should I wait until the bug is fixed to continue with this one or use the `EndsWith` alternative?

**Comment by Haacked** [comment]
Use the `EndsWith`. It'll be future proof. Hopefully.

File: Octokit/Clients/RepositoriesClient.cs [hunk 1]
```diff
@@ -116,6 +116,12 @@ public Task<Repository> Create(string organizationLogin, NewRepository newReposi
                 {
                     throw new PrivateRepositoryQuotaExceededException(e);
                 }
+                
+                if (errorMessage != null && errorMessage.EndsWith("is an unknown gitignore template.", StringComparison.OrdinalIgnoreCase))
```
Comment (shiftkey) [inline Octokit/Clients/RepositoriesClient.cs]: Could we clean this up so it looks like the `String.Equals` call above?

Comment (shiftkey) [inline Octokit/Clients/RepositoriesClient.cs]: I might have been too eager with this, given you need `EndsWith` to verify this. Disregard.

**Comment by shiftkey** [comment]
Thanks for this - I merged in the `csproj` conflict so this is :ship:ed!

**Pull request merged**